lib/sqlalchemy/sql/functions.py

```diff
@@ -49,7 +56,38 @@ def register_function(identifier, fn, package="_default"):
 
     """
     reg = _registry[package]
-    reg[identifier] = fn
+    raw_identifier = identifier
+    identifier = identifier.lower()
+
+    # Check if a function with the same lowercase identifier is registered.
+    if identifier in _case_sensitive_reg[package]:
```

Comment:
**mike bayer** (zzzeek) wrote:

so all the _case_sensitive_reg[package] lookups lets do them just once above, see where it says "reg = _registry[package]", add another one, "case_sensitive_reg = _case_sensitive_registry[package]" (this is why I went with "registry" and not "reg" above).   that should reduce overhead / verbosity in the rest of this.

Reply:
You are right, it will be far better.